agents/lakeshore372/LS372_agent.py

```diff
@@ -977,6 +985,79 @@ def get_still_output(self, session, params=None):
 
         return True, "Current still output is {}".format(still_output)
 
+    def input_configfile(self, configfile, filepath='/ls372configs/'):
+        """ Upload 372 configuration file to initialize channel/device
+        settings upon Agent startup.
+
+        Args:
+            configfile (str): name of .yaml config file
+        """
+        # include /ls372configs/ dir where configfiles are in docker container
+        ls372config = os.path.join(filepath, configfile)
+        with open(ls372config) as f:
+            config = yaml.safe_load(f)
+
+        ls = self.module
+        ls_id = self.module.id
+        ls_info = ls_id.split(',')
+        ls_serial = ls_info[2]
+
+        # TODO: assert 372 instance id == serial num?
+
+        device_config = config[ls_serial]['device_settings']
+        ls_chann_settings = config[ls_serial]['channel']
+
+        # enable/disable autoscan
+        if device_config['autoscan'] == 'on':
+            ls.enable_autoscan()
+            print('Autoscan enabled')
+        elif device_config['autoscan'] == 'off':
+            ls.disable_autoscan()
+            print('Autoscan disabled')
+
+        for i in ls_chann_settings:
+            lschann = Channel(self.module, i)
```

Comment:
The `LS372` object already has objects for each `Channel` initialized, so I don't think you should reinitialize new `Channel` objects here, but instead use those already in the `ls` instance here.